File: Leetcode-one-by-one/src/Maximum Total Subarray Value I.swift
class Solution {
    func maxTotalValue(_ nums: [Int], _ k: Int) -> Int64 {
        var mn = Int.max
        var mx = Int.min
        
        for x in nums {
            mn = min(mn, x)
            mx = max(mx, x)
        }
        
        return Int64(k) * Int64(mx - mn)
    }
}

File: Leetcode-one-by-one/src/Maximum Total Subarray Value I.go
func maxTotalValue(nums []int, k int) int64 {
    mn, mx := nums[0], nums[0]

    for _, x := range nums {
        if x < mn {
            mn = x
        }
        if x > mx {
            mx = x
        }
    }

    return int64(k) * int64(mx-mn)
}
